src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -783,44 +780,22 @@ internal static bool TryFormatBigInteger(BigInteger value, ReadOnlySpan<char> fo
                 }
             }
 
-            // First convert to base 10^9.
-            int cuSrc = value._bits.Length;
-            // A quick conservative max length of base 10^9 representation
-            // A uint contributes to no more than 10/9 of 10^9 block, +1 for ceiling of division
-            int cuMax = cuSrc * (MaxPartialDigits + 1) / MaxPartialDigits + 1;
-            Debug.Assert((long)BigInteger.MaxLength * (MaxPartialDigits + 1) / MaxPartialDigits + 1 < (long)int.MaxValue); // won't overflow
-
-            uint[]? bufferToReturn = null;
-            Span<uint> base1E9Buffer = cuMax < BigIntegerCalculator.StackAllocThreshold ?
-                stackalloc uint[cuMax] :
-                (bufferToReturn = ArrayPool<uint>.Shared.Rent(cuMax));
-
-            int cuDst = 0;
+            // The Ratio is calculated as: log_{10^9}(2^32)
+            const double digitRatio = 1.0703288734719332;
+            Debug.Assert(BigInteger.MaxLength * digitRatio + 1 < Array.MaxLength); // won't overflow
 
-            for (int iuSrc = cuSrc; --iuSrc >= 0;)
-            {
-                uint uCarry = value._bits[iuSrc];
-                for (int iuDst = 0; iuDst < cuDst; iuDst++)
-                {
-                    Debug.Assert(base1E9Buffer[iuDst] < TenPowMaxPartial);
+            int base1E9BufferLength = (int)(value._bits.Length * digitRatio) + 1;
+            uint[]? base1E9BufferFromPool = null;
+            Span<uint> base1E9Buffer = base1E9BufferLength < BigIntegerCalculator.StackAllocThreshold ?
```

Comment:
Add an `(uint)` cast to check lower bound of `base1E9BufferLength` (and make it clear that both bounds are checked)

Reply:
I added (uint). Also, to match other parts, I used <= instead of < and removed the excess portion using `Slice`.